src/Goteo/Repository/DataSetRepository.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/*
+ * This file is part of the Goteo Package.
+ *
+ * (c) Platoniq y Fundación Goteo <fundacion@goteo.org>
+ *
+ * For the full copyright and license information, please view the README.md
+ * and LICENSE files that was distributed with
+ */
+
+ namespace Goteo\Repository;
+
+ use Goteo\Application\Exception\ModelException;
+ use Goteo\Application\Exception\ModelNotFoundException;
+ use Goteo\Model\DataSet;
+ use PDO;
+ use PDOException;
+
+ class DataSetRepository extends BaseRepository {
+     protected ?string $Table = 'data_set';
+
+     public function getById(int $id): DataSet  {
+         $sql = "SELECT data_set.*
+                FROM data_set
+                WHERE data_set.id = ?";
+
+         $dataSet = $this->query($sql, [$id])->fetchObject(DataSet::class);
+
+         if (!$dataSet instanceOf DataSet)
+             throw new ModelNotFoundException("DataSet with id $id not found");
+
+         return $dataSet;
+    }
+
+    public function getList(array $filter, int $offset = 0, int $limit = 10, bool $count = false) {
+         if ($count) {
+             $sql = "SELECT count(data_set.*)
+                FROM data_set
+                LIMIT $limit
+                OFFSET $offset";
+
+             return $this->query($sql)->fetchAll();
+         }
+
+         $sql = "SELECT data_set.*
+                FROM data_set
+                LIMIT $limit
+                OFFSET $offset";
+
+        return $this->query($sql)->fetchAll(PDO::FETCH_CLASS, DataSet::class );
+    }
+
+    public function getListByFootprint(array $footprints) {
+
+         $sqlWhere = "";
+         if (!empty($footprints)) {
+            $sqlWhere = "WHERE fds.footprint_id IN ( " . implode(',', $footprints) . ")";
+         }
+
+        $sql = "SELECT data_set.*
+                FROM data_set
+                INNER JOIN footprint_data_set fds ON fds.data_set_id = data_set.id
+                {$sqlWhere}
+                ";
+
+        return $this->query($sql)->fetchAll(PDO::FETCH_CLASS, DataSet::class );
```

Comment:
@davidbeig are we returning an array of DataSet objects here? If so, we could add in the PHPDoc of this method, the following:

```
/**
* @return DataSet[]
**/
```

This way, the IDE will recognize that we are returning an array of DataSet, instead of a simple array.